### PR TITLE
Add e2e options to set CNI hub/tag

### PR DESCRIPTION
**Pull request by john-a-joyce**
This change adds the ability to override the
hub and tag value produced by the generate_yaml
operation.

### CODE REVIEW

File: tests/e2e/README.md [hunk 1]
```diff
@@ -107,6 +107,8 @@ E2E tests have multiple options available while running them as follows:
 * `--sidecar_injector_hub <hub>` - Image hub for the Sidecar Injector (default: environment $HUB)
 * `--sidecar_injector_tag <tag>` - Image tag for the Sidecar Injector (default: environment $TAG)
 * `--sidecar_injector_file <file>` - Sidecar injector YAML file name (default: istio-sidecar-injector.yaml)
+* `--cni_hub <hub>` - Image hub for the CNI plugin: (default: environment $HUB)
```
Comment (baodongli) [inline tests/e2e/README.md]: Consider making it a bit specific such as:
--istio_cni_hub <hub> - Image hub for the Istio CNI plugin: ....
Similar comments with cni_tag.

File: tests/istio.mk [hunk 2]
```diff
@@ -57,11 +57,13 @@ DEFAULT_EXTRA_E2E_ARGS += --pilot_tag=${TAG}
 DEFAULT_EXTRA_E2E_ARGS += --proxy_tag=${TAG}
 DEFAULT_EXTRA_E2E_ARGS += --ca_tag=${TAG}
 DEFAULT_EXTRA_E2E_ARGS += --galley_tag=${TAG}
+DEFAULT_EXTRA_E2E_ARGS += --cni_tag=${TAG}
```
Comment (tiswanso) [inline tests/istio.mk]: remove this.  Let the helm chart default it.

File: tests/istio.mk [hunk 3]
```diff
@@ -57,11 +57,13 @@ DEFAULT_EXTRA_E2E_ARGS += --pilot_tag=${TAG}
...
 DEFAULT_EXTRA_E2E_ARGS += --mixer_hub=${HUB}
 DEFAULT_EXTRA_E2E_ARGS += --pilot_hub=${HUB}
 DEFAULT_EXTRA_E2E_ARGS += --proxy_hub=${HUB}
 DEFAULT_EXTRA_E2E_ARGS += --ca_hub=${HUB}
 DEFAULT_EXTRA_E2E_ARGS += --galley_hub=${HUB}
+DEFAULT_EXTRA_E2E_ARGS += --cni_hub=${HUB}
```
Comment (tiswanso) [inline tests/istio.mk]: remove this.  Let the helm chart default it.

File: tests/e2e/framework/kubernetes.go [hunk 4]
```diff
@@ -87,6 +87,8 @@ var (
 	galleyTag          = flag.String("galley_tag", os.Getenv("TAG"), "Galley tag")
 	sidecarInjectorHub = flag.String("sidecar_injector_hub", os.Getenv("HUB"), "Sidecar injector hub")
 	sidecarInjectorTag = flag.String("sidecar_injector_tag", os.Getenv("TAG"), "Sidecar injector tag")
+	cniHub             = flag.String("cni_hub", os.Getenv("HUB"), "CNI hub")
```
Comment (tiswanso) [inline tests/e2e/framework/kubernetes.go]: remove this and line below. Let the helm chart default it.

File: tests/istio.mk (hunk 3 above)
Comment (tiswanso) [reply tests/istio.mk]: NOTE:  these actually conflict if you try to set with E2E_ARGS

File: tests/e2e/framework/kubernetes.go (hunk 4 above)
Comment (john-a-joyce) [reply tests/e2e/framework/kubernetes.go]: I removed the default value (actually set it to ""), but I didn't remove the line.

**Comment by john-a-joyce** [comment]
This might not be needed so closing for now

**Pull request closed without merging**